splash/tests/test_execute_element.py

```diff
@@ -0,0 +1,1416 @@
+
+    def test_remove_event_listeners_bad_event_name(self):
+        resp = self.request_lua("""
+        function main(splash)
+            assert(splash:go(splash.args.url))
+            assert(splash:wait(0.1))
+
+            local body = splash:select('body')
+            body.node:addEventListener('click', function(event) end)
+            body.node:removeEventListener('', function(event) end)
+
+            return true
+        end
+        """, {"url": self.mockurl("various-elements")})
+
+        err = self.assertScriptError(resp, ScriptError.SPLASH_LUA_ERROR,
+                                     message='event_name must be specified')
+        self.assertEqual(err['info']['splash_method'], 'removeEventListener')
+
+    def test_remove_not_added_event_listeners(self):
+        resp = self.request_lua("""
+        function main(splash)
+            assert(splash:go(splash.args.url))
+            assert(splash:wait(0.1))
+
+            local body = splash:select('body')
+            body.node:removeEventListener('click', function(event) end)
+
+            return true
+        end
+        """, {"url": self.mockurl("various-elements")})
+
+        self.assertStatusCode(resp, 200)
+        self.assertEqual(resp.text, 'True')
+
+    def test_submit(self):
+        resp = self.request_lua("""
+        function main(splash)
+            assert(splash:go(splash.args.url))
+            assert(splash:wait(0.1))
+
+            local submitted = false
+
+            local form = splash:select('form')
+
+            assert(form:submit())
+            assert(splash:wait(0.5))
+
+            return splash:url()
+        end
+        """, {"url": self.mockurl("various-elements")})
+
+        self.assertStatusCode(resp, 200)
+        self.assertRegexpMatches(resp.text, '/submitted\?username=admin&password=pass123')
+
+    def test_submit_not_form(self):
+        resp = self.request_lua("""
+        function main(splash)
+            assert(splash:go(splash.args.url))
+            assert(splash:wait(0.1))
+
+            local submitted = false
+
+            local input = splash:select('input')
+
+            assert(input:submit())
+        end
+        """, {"url": self.mockurl("various-elements")})
+
+        self.assertScriptError(resp, ScriptError.LUA_ERROR, "Node should be 'form'")
+
+    def test_element_arg(self):
+        resp = self.request_lua("""
+        function main(splash)
+            assert(splash:go(splash.args.url))
+            assert(splash:wait(0.1))
+
+            local body = splash:select('body')
+            local div = splash:evaljs('document.createElement("div")')
+
+            div.node.id = 'mydiv';
+            body.node:appendChild(div);
+
+            return body.node.lastChild.node.id
+        end
+        """, {"url": self.mockurl("various-elements")})
+
+        self.assertStatusCode(resp, 200)
+        self.assertEqual(resp.text, 'mydiv')
+
+    def test_field_value_emtpy(self):
+        pass
```

Comment:
I've added some tests (see the last two), but  I don't understand what do you want.

>the second checks that we work on element instances and not e.g. store unique css selectors
like if we change the refresh the page and try to access some property and get error, right?
>3rd test checks that we don't rely on splash:go and other methods to clear elements state
should I check that I'm getting error on various type of page change (js redrect, splash:go)?